dask/base.py

```diff
@@ -416,14 +416,32 @@ def tokenize(*args, **kwargs):
     return md5(str(tuple(map(normalize_token, args))).encode()).hexdigest()
 
 
+def dont_optimize(dsk, keys):
+    return dsk
+
+
+def optimization_function(obj):
```

Comment:
We could take this further and move this decorator to work on any function, not just optimizations.

```python
from .context import _globals, defer_to_override

@staticmethod
@defer_to_override('array_optimize', falsey=dont_optimize)
def _optimize(dsk, keys):
    ...
```

I think we could improve on the name falsey.  This feels slightly better to me because we're not creating an override system just for optimizations, but presumably for a variety of parts of the codebase.  Of course this doesn't mean that we should use this system, just that if we find ourselves in this position again we'll be able to reuse the existing system and API.